flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/fanout/FanOutProperties.java

```diff
@@ -0,0 +1,127 @@
+package org.apache.flink.streaming.connectors.kinesis.internals.fanout;
+
+import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants;
+import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants.EFORegistrationType;
+import org.apache.flink.streaming.connectors.kinesis.config.ConsumerConfigConstants.RecordPublisherType;
+import org.apache.flink.streaming.connectors.kinesis.util.KinesisConfigUtil;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+
+/**
+ * This is a configuration class for enhanced fan-out components.
+ */
+public class FanOutProperties implements Serializable {
+	private static final long serialVersionUID = 3204635913413261619L;
+
+	private EFORegistrationType efoRegistrationType;
+	@Nullable
+	private String consumerName;
+	@Nullable
+	private List<String> streamConsumerArns;
```

Comment:
@xiaolong-sn having a `List` means we lose the Stream > ARN relationship. Can you use a `Map` instead where the key is stream and value is ARN?

Reply:
Yes, I replace the `List` with `Map` now.